these tests rely on CMP0048 OLD behavior which has been
removed from cmake 4 so they can never pass

also work around jankery in ECMGeneratePkgConfigFile
which has the same root cause but i'm not sure how to
properly fix it

--- a/tests/ECMGeneratePkgConfigFile/run_test.cmake.config
+++ b/tests/ECMGeneratePkgConfigFile/run_test.cmake.config
@@ -40,6 +40,7 @@ endfunction()
 
 ###########################################################
 
+set(PROJECT_VERSION 5.43)
 
 ecm_setup_version(5.43 VARIABLE_PREFIX KCOREADDONS
                 VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/kcoreaddons_version.h"
@@ -143,6 +143,8 @@ message(STATUS "Test7: Custom defines")
 set(origfiles "${CMAKE_CURRENT_SOURCE_DIR}/KF6Attica.pc")
 set(prefix "/usr")
 
+set(PROJECT_VERSION 6.7.0)
+
 ecm_setup_version(6.7.0 VARIABLE_PREFIX ATTICA
                 VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/attica_version.h"
                 PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF6AtticaConfigVersion.cmake"
--- a/tests/ECMSetupVersionTest/CMakeLists.txt
+++ b/tests/ECMSetupVersionTest/CMakeLists.txt
@@ -13,17 +13,6 @@ macro(add_version_test NAME COMMAND)
     --test-command ${COMMAND} ${ARGN})
 endmacro()
 
-add_version_test(old_simple dummy)
-add_version_test(old_simple_no_version_string_vars dummy)
-add_version_test(old_soversion dummy)
-add_version_test(old_version_file dummy)
-add_version_test(old_version_file_abspath dummy)
-add_version_test(old_version_file_anynewer dummy)
-add_version_test(old_version_file_exact dummy)
-add_version_test(old_version_file_samemajor dummy)
-add_version_test(old_header check_header)
-add_version_test(old_header_abspath check_header)
-
 add_version_test(new_explicit_header check_header)
 add_version_test(new_explicit_header_abspath check_header)
 add_version_test(new_explicit_simple dummy)
